src/components/date-range-picker/date-range-picker.ts

```diff
@@ -0,0 +1,1211 @@
+   * Determines whether the calendar is opened in a dropdown or a modal dialog
+   * @attr mode
+   */
+  @property()
+  public mode: PickerMode = 'dropdown';
+
+  /**
+   * Use two inputs to display the date range values. Makes the input editable in dropdown mode.
+   * @attr use-two-inputs
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'use-two-inputs' })
+  public useTwoInputs = false;
+
+  /**
+   * Whether the control will show chips with predefined ranges.
+   * @attr
+   */
+  @property({
+    reflect: true,
+    type: Boolean,
+    attribute: 'use-predefined-ranges',
+  })
+  public usePredefinedRanges = false;
+
+  /**
+   * The locale settings used to display the value.
+   * @attr
+   */
+  @property()
+  public locale = 'en';
+
+  /** The resource strings of the date range picker. */
+  @property({ attribute: false })
+  public resourceStrings: IgcDateRangePickerResourceStrings =
+    IgcDateRangePickerResourceStringsEN;
+
+  // #region Input-related properties
+
+  /**
+   * Makes the control a readonly field.
+   * @attr readonly
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'readonly' })
+  public readOnly = false;
+
+  /**
+   * Whether to allow typing in the input.
+   * @attr non-editable
+   */
+  @property({ type: Boolean, reflect: true, attribute: 'non-editable' })
+  public nonEditable = false;
+
+  /**
+   * Whether the control will have outlined appearance.
+   * @attr
+   */
+  @property({ reflect: true, type: Boolean })
+  public outlined = false;
+
+  /**
+   * The label of the control (single input).
+   * @attr label
+   */
+  @property()
+  public label!: string;
+
+  /**
+   * The label attribute of the start input.
+   * @attr label-start
+   */
+  @property({ attribute: 'label-start' })
+  public labelStart = '';
+
+  /**
+   * The label attribute of the end input.
+   * @attr label-end
+   */
+  @property({ attribute: 'label-end' })
+  public labelEnd = '';
+
+  /**
+   * The placeholder attribute of the control (single input).
+   * @attr
+   */
+  @property()
+  public set placeholder(value: string) {
+    this._placeholder = value;
+  }
+
+  public get placeholder(): string {
+    const rangePlaceholder = `${this.inputFormat} - ${this.inputFormat}`;
+    return this._placeholder ?? rangePlaceholder;
+  }
+
+  /**
+   * The placeholder attribute of the start input.
+   * @attr placeholder-start
+   */
+  @property({ attribute: 'placeholder-start' })
+  public placeholderStart = '';
+
+  /**
+   * The placeholder attribute of the end input.
+   * @attr placeholder-end
+   */
+  @property({ attribute: 'placeholder-end' })
+  public placeholderEnd = '';
+
+  /** The prompt symbol to use for unfilled parts of the mask.
+   *  @attr
+   */
+  @property()
+  public prompt = '_';
+
+  /**
+   * Format to display the value in when not editing.
+   * Defaults to the input format if not set.
+   * @attr display-format
+   */
+  @property({ attribute: 'display-format', reflect: false })
```

Comment:
```suggestion
  @property({ attribute: 'display-format' })
```
Lit does not reflect attributes by default